js/views/modals/orderDetail/summaryTab/Summary.js

```diff
@@ -446,6 +478,225 @@ export default class extends BaseVw {
+    // so in the escrow timeouts 45 is represented as '45 days' instead of '1 month'.
+    moment.relativeTimeThreshold('d', 364);
+
+    const height = app.walletBalance.get('height');
+    let state = {
+      ownPeerId: app.profile.id,
+      buyer: this.buyer.id,
+      vendor: this.vendor.id,
+      moderator: this.moderator && this.moderator.id || undefined,
+      awaitingBlockHeight: false,
+      isFundingConfirmed: false,
+      blockTime: cryptoCur.blockTime,
+      isCompletable: orderState === 'FULFILLED',
+      isDisputed: orderState === 'DISPUTED',
+      hasDisputeEscrowExpired: false,
+      isPaymentClaimable: false,
+      isPaymentFinalized: false,
+      showDisputeBtn: false,
+      showDiscussBtn: orderState === 'DISPUTED',
+      showResolveDisputeBtn: false,
+    };
+
+    if (!height) {
+      // temporary, this will not be needed once this server issue is completed:
+      // https://github.com/OpenBazaar/openbazaar-go/issues/843
+      state.awaitingBlockHeight = true;
+    } else if (orderState === 'PAYMENT_FINALIZED') {
+      state.isPaymentFinalized = true;
+    } else {
+      let disputeStartTime;
+      let escrowTimeoutHours;
+
+      try {
+        escrowTimeoutHours = this.contract.escrowTimeoutHours;
+      } catch (e) {
+        // pass - will be handled below
+      }
+
+      if (orderState === 'DISPUTED') {
+        try {
+          disputeStartTime = this.isCase() ?
+            this.model.get('timestamp') :
+            this.contract.get('dispute').timestamp;
+        } catch (e) {
+          // pass - will be handled below
+        }
+      }
+
+      if (
+        !escrowTimeoutHours ||
+        (orderState === 'DISPUTED' && !Date.parse(disputeStartTime))
+      ) {
+        // contract probably forged
+        state = {
+          ...state,
+          invalidContractData: true,
+          showDisputeBtn: this.isOrderStateDisputable,
+          showResolveDisputeBtn: this.isCase(),
+        };
+      } else {
+        let hasDisputeEscrowExpired;
+        state.totalTime = this.escrowTimeoutHoursVerbose;
+
+        if (this.isCase() || orderState === 'DISPUTED') {
+          const msSinceDisputeStart = Date.now() - (new Date(disputeStartTime)).getTime();
+          const msRemaining = (escrowTimeoutHours * 60 * 60 * 1000) -
+            msSinceDisputeStart;
+          hasDisputeEscrowExpired = msRemaining <= 0;
+
+          state = {
+            ...state,
+            hasDisputeEscrowExpired,
+            timeRemaining: hasDisputeEscrowExpired ? 0 :
+              moment(Date.now()).from(moment(Date.now() + msRemaining), true),
+            showDiscussBtn: !hasDisputeEscrowExpired,
+          };
+
+          if (!hasDisputeEscrowExpired) {
+            let checkBackInMs = 1000; // every second
+
+            if (msRemaining > 1000 * 60 * 60 * 24) {
+              // greater than a day
+              checkBackInMs = 1000 * 60 * 60 * 20;
```

Comment:
Should 20 be 24 here?

Reply:
The JS timeout and interval functions are not super accurate, so I'm purposely undershooting it.   So it will end up coming back to early and then lock into progressively more frequent checking intervals.